File: skip-the-dishes-domain/src/main/java/br/com/skip/the/dishes/domain/order/elements/Handler.java
package br.com.skip.the.dishes.domain.order.elements;

import br.com.zup.eventsourcing.core.AggregateId;

public interface Handler {

    void on(OrderCreated orderCreated);

    void on(AggregateId aggregateId, ProductAdded productAdded);

    void on(AggregateId aggregateId, ProductDeleted productDeleted);

    void on(AggregateId aggregateId, OrderCancelled orderCancelled);

    void on(AggregateId aggregateId, OrderRequested orderRequested);

    void on(AggregateId aggregateId, DeliveryAddressUpdated deliveryAddressUpdated);

}

File: skip-the-dishes-domain/src/main/java/br/com/skip/the/dishes/domain/order/elements/Handler.kt
package br.com.skip.the.dishes.domain.order.elements

import br.com.zup.eventsourcing.core.AggregateId

interface Handler {

    fun on(orderCreated: OrderCreated)

    fun on(aggregateId: AggregateId, productAdded: ProductAdded)

    fun on(aggregateId: AggregateId, productDeleted: ProductDeleted)

    fun on(aggregateId: AggregateId, orderCancelled: OrderCancelled)

    fun on(aggregateId: AggregateId, orderRequested: OrderRequested)

    fun on(aggregateId: AggregateId, deliveryAddressUpdated: DeliveryAddressUpdated)

}
